src/overlays/actors/ovl_En_Skjneedle/z_en_skjneedle.c

```diff
@@ -27,13 +29,70 @@ const ActorInit En_Skjneedle_InitVars = {
     (ActorFunc)EnSkjneedle_Update,
     (ActorFunc)EnSkjneedle_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/EnSkjneedle_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/EnSkjneedle_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skjneedle/func_80B01F6C.s")
+static ColliderCylinderInit_Set3 sCylinderInit = 
+{
+	{ COLTYPE_UNK1, 0x11, 0x09, 0x00, COLSHAPE_CYLINDER },
+	{ 0x00, { 0xFFCFFFFF, 0x00, 0x08, }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x01, 0x01, 0x01 },
+	{ 10, 4, -2, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+	ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+	ICHAIN_F32(unk_4C, 30, ICHAIN_STOP)
+};
```

Comment:
to keep clang from messing up formatting you should leave a trailing comma on the last entry. this also means you should run clang. youll need to install clang-format and clang-tidy then run format.sh which is in the root directory